gnssanalysis/gn_download.py

```diff
@@ -846,325 +720,190 @@ def download_multiple_files_from_cddis(files: [str], ftp_folder: str, output_fol
-                            f_uncomp = gen_uncomp_filename(f)
-                            if f_uncomp not in f_dict[f_typ]:
-                                f_dict[f_typ].append(f_uncomp)
-    if f_dict:
-        return f_dict
-
-
-def download_pea_prods(
-    dest,
-    most_recent=True,
-    dates=None,
-    ac="igs",
-    out_dict=False,
-    trop_vmf3=False,
-    brd_typ="igs",
-    snx_typ="igs",
-    clk_sel="clk",
-    repro3=False,
-):
-    """
-    Download necessary pea product files for date/s provided
-    """
-    if dest[-1] != "/":
-        dest += "/"
-
-    if most_recent:
-        snx_vars_out = download_most_recent(
-            dest=dest, f_type="snx", ac=snx_typ, dwn_src="cddis", f_dict_out=True, gpswkD_out=True, ftps_out=True
-        )
-        f_dict, gpswkD_out, ftps = snx_vars_out
-
-        clk_vars_out = download_most_recent(
-            dest=dest, f_type=clk_sel, ac=ac, dwn_src="cddis", f_dict_out=True, gpswkD_out=True, ftps_out=True
+                product_filename, gps_date, reference_start = generate_product_filename(
+                    reference_start,
+                    file_ext,
+                    shift=24,  # Shift at the start of the loop - speeds up total download time
+                    long_filename=long_filename,
+                    analysis_center=analysis_center,
+                    timespan=timespan,
+                    solution_type=solution_type,
+                    sampling_rate=sampling_rate,
+                    project=project_type,
+                )
+                download_filepath = check_whether_to_download(
+                    filename=product_filename, download_dir=download_dir, if_file_present=if_file_present
+                )
+                if download_filepath:
+                    download_file_from_cddis(
+                        filename=product_filename,
+                        ftp_folder=f"gnss/products/{gps_date.gpswk}",
+                        output_folder=download_dir,
+                        if_file_present=if_file_present,
+                        note_filetype=file_ext,
+                    )
+                count += 1
+                remain = end_epoch - reference_start
+
+
+def download_atx(download_dir: _Path, reference_frame: str = "IGS20", if_file_present: str = "prompt_user") -> _Path:
+    """Download the ATX file necessary for running the PEA provided the download directory (download_dir)
+
+    :param _Path download_dir: Where to download files (local directory)
+    :param str reference_frame: Coordinate reference frame file to download, defaults to "IGS20"
+    :param str if_file_present: What to do if file already present: "replace", "dont_replace", defaults to "prompt_user"
+    :return _Path: The pathlib.Path of the downloaded file
+    """
+    match reference_frame:
+        case "IGS20":
+            atx_filename = "igs20.atx"
+        case "IGb14":
+            atx_filename = "igs14.atx"
```

Comment:
This should perhaps raise a `ValueError` as a default, indicating that the frame name provided was not accepted.

Reply:
Done